zerver/views/users.py

```diff
@@ -425,6 +426,7 @@ def patch_bot_backend(
 
     json_result = dict(
         full_name=bot.full_name,
+        avatar_source=avatar_source,
```

Comment:
I don't think it is necessary to pass `avatar_source` here in the result dict. We don't do so for human users as well.

Reply:
I was passing this in the result just to update avatar_source in the frontend. If avatar_source updates in the frontend in real-time, this won't be needed anymore.

Reply:
Ok, Maybe you can try the solution mentioned [here](https://github.com/zulip/zulip/pull/24697#issuecomment-1482536795).